cs-50/week-3/tideman.c

```diff
@@ -177,28 +175,37 @@ void add_pairs(void)
     return;
 }
 
-void shuffle_pairs_for_fun(void)
-{
-    for(int i=0; i<pair_count; i++){
-        int random_index = (int) arc4random_uniform(i + 1);
-        pair original_pair_at_index = pairs[random_index];
-        pairs[random_index] = pairs[i];
-        pairs[i] = original_pair_at_index;
-    }
-    return;
-}
-
 // Sort pairs in decreasing order by strength of victory
 void sort_pairs(void)
 {
-    
+    for (int i = 0; i < pair_count-1; i++) {
+        for (int j = i+1; j < pair_count; j++) {
+            int lead = preferences[pairs[j].winner][pairs[j].loser];
+            if (preferences[pairs[i].winner][pairs[i].loser] < lead) {
+                pair tmp = pairs[i];
+                pairs[i] = pairs[j];
+                pairs[j] = tmp;
+        }
+        }
+    }
```

Comment:
Wow, how does it sort even?

Reply:
I added the shuffle back in too with more candidates/voters and it works!

Reply:
Is it a bubble sort? o:

Reply:
Yes(-ish)! Instead of passing through the list and comparing neighboring elements, `i` and `i+1`, this compares one element at `i` with all other elements in the list, `i+n`, with each pass—which google tells me is an exchange sort? Bubble sort's cousin?

Reply:
Oh I see you are right, exchanges all elems one by one with the most left finding the min elem